src/SeaIceDynamics/Rheologies/explicit_visco_plastic_rheology.jl

```diff
@@ -0,0 +1,278 @@
+                                        convert(FT, min_substeps),
+                                        convert(FT, max_substeps))
+end
+
+function required_auxiliary_fields(grid, ::ExplicitViscoPlasticRheology)
+    
+    # TODO: What about boundary conditions?
+    σ₁₁ = Field{Center, Center, Nothing}(grid)
+    σ₂₂ = Field{Center, Center, Nothing}(grid)
+    σ₁₂ = Field{Face, Face, Nothing}(grid)
+    
+    uⁿ = Field{Face, Center, Nothing}(grid)
+    vⁿ = Field{Center, Face, Nothing}(grid)
+    P  = Field{Center, Center, Nothing}(grid)
+
+    substeps = Field{Face, Face, Nothing}(grid) # Dynamic substeps a la Kimmritz et al (2016)
+
+    # An initial (safe) educated guess
+    fill!(substeps, 100)
+
+    return (; σ₁₁, σ₂₂, σ₁₂, substeps, uⁿ, vⁿ, P)
+end
+
+# Extend the `adapt_structure` function for the ExplicitViscoPlasticRheology
+Adapt.adapt_structure(to, r::ExplicitViscoPlasticRheology) = 
+    ExplicitViscoPlasticRheology(Adapt.adapt(to, r.ice_compressive_strength),
+                                 Adapt.adapt(to, r.ice_compaction_hardening),
+                                 Adapt.adapt(to, r.yield_curve_eccentricity),
+                                 Adapt.adapt(to, r.Δ_min),
+                                 Adapt.adapt(to, r.min_substeps),
+                                 Adapt.adapt(to, r.max_substeps))
+
+@inline function fill_rheology_halo_regions!(solver, ::ExplicitViscoPlasticRheology) 
+    fill_halo_regions!(solver.auxiliary_fields.σ₁₁)
+    fill_halo_regions!(solver.auxiliary_fields.σ₁₂)
+    fill_halo_regions!(solver.auxiliary_fields.σ₂₂)
+    fill_halo_regions!(solver.auxiliary_fields.substeps)
+
+    return nothing
+end
+
+"""
+    initialize_rheology!(model, rheology::ExplicitViscoPlasticRheology)
+
+Initialize the elasto-visco-plastic rheology.
+In this step we calculate the ice strength given the ice mass (thickness and concentration).
+"""
+function initialize_rheology!(model, rheology::ExplicitViscoPlasticRheology)
+    h = model.ice_thickness
+    ℵ = model.ice_concentration
+
+    P★ = rheology.ice_compressive_strength
+    C  = rheology.ice_compaction_hardening
+    
+    u, v   = model.velocities
+    fields = model.ice_dynamics.auxiliary_fields
+
+    # compute on the whole grid including halos
+    parameters = KernelParameters(size(fields.P.data)[1:2], fields.P.data.offsets[1:2])
+    launch!(architecture(model.grid), model.grid, parameters, _initialize_evp_rhology!, fields, grid, P★, C, h, ℵ, u, v)
+    
+    return nothing
+end
+
+@kernel function _initialize_evp_rhology!(fields, grid, P★, C, h, ℵ, u, v)
+    i, j = @index(Global, NTuple)    
+    @inbounds fields.P[i, j, 1]  = ice_strength(i, j, 1, grid, P★, C, h, ℵ)
+    @inbounds fields.uⁿ[i, j, 1] = u[i, j, 1]
+    @inbounds fields.vⁿ[i, j, 1] = v[i, j, 1]
+end
+
+# The parameterization for an `ExplicitViscoPlasticRheology`
+@inline ice_strength(i, j, k, grid, P★, C, h, ℵ) = P★ * h[i, j, k] * exp(- C * (1 - ℵ[i, j, k])) 
```

Comment:
this is the one thing that we actually want to have a struct parameterization for